idl/testing/run_tests.py

```diff
@@ -0,0 +1,1218 @@
+
+mdsdisconnect, socket=0
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+if (test_status eq PASS) then begin
+    print, 'SUCCESS'
+end else begin
+    print, 'FAILURE'
+endelse
+
+''',
+'''
+
+% Compiled module: MDSCONNECT.
+% Compiled module: MDS_KEYWORD_SET.
+% Compiled module: MDSDISCONNECT.
+% Compiled module: MDSVALUE.
+% Compiled module: MDSCHECKARG.
+% Compiled module: MDSISCLIENT.
+result=          55
+result=          55
+SUCCESS
+
+''')
+
+
+# Issue #2639: mdsvalue and kill default socket.
+idl_test(f'''
+         
+testid = 'IDL-2639-kill-default-socket'
+PASS = 1
+FAIL = 0
+DATA = '55'
+mdsip_server = '{server}'
+test_status = PASS
+
+mdsconnect, mdsip_server
+mdsconnect, mdsip_server
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+mdsdisconnect
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+if (test_status eq PASS) then begin
+    print, 'SUCCESS'
+end else begin
+    print, 'FAILURE'
+endelse
+
+''',
+'''
+
+% Compiled module: MDSCONNECT.
+% Compiled module: MDS_KEYWORD_SET.
+% Compiled module: MDSDISCONNECT.
+% Compiled module: MDSVALUE.
+% Compiled module: MDSCHECKARG.
+% Compiled module: MDSISCLIENT.
+result=          55
+% Compiled module: MDSIDLIMAGE.
+result=          55
+SUCCESS
+
+''')
+
+
+# Issue #2639: mdsvalue, one connect, and kill socket.
+idl_test(f'''
+         
+testid = 'IDL-2639-kill-single-socket'
+PASS = 1
+FAIL = 0
+DATA = '55'
+mdsip_server = '{server}'
+test_status = PASS
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+mdsconnect, mdsip_server
+mdsdisconnect
+
+result = mdsvalue(DATA)
+print, 'result=', result
+if (result ne fix(DATA)) then test_status = FAIL
+
+if (test_status eq PASS) then begin
+    print, 'SUCCESS'
+end else begin
+    print, 'FAILURE'
+endelse
+
+''',
+'''
+
+% Compiled module: MDSVALUE.
+% Compiled module: MDSCHECKARG.
+% Compiled module: MDSISCLIENT.
+% Compiled module: MDS_KEYWORD_SET.
+% Compiled module: MDSIDLIMAGE.
+result=          55
+% Compiled module: MDSCONNECT.
+% Compiled module: MDSDISCONNECT.
+result=          55
+SUCCESS
+
+''')
+
+
+# Issue #2639: mdsvalue, one connect, and kill 0 socket.
+# Note different behavior from IDL-2639-kill-single-socket test.
+# idl_test(f'''
+         
+# testid = 'IDL-2639-kill-single-zero'
+# PASS = 1
+# FAIL = 0
+# DATA = '55'
+# mdsip_server = '{server}'
+# test_status = PASS
+
+# result = mdsvalue(DATA)
+# print, 'result=', result
+# if (result ne fix(DATA)) then test_status = FAIL
+
+# mdsconnect, mdsip_server
+# mdsdisconnect, socket=0
+
+# result = mdsvalue(DATA)
+# print, 'result=', result
+# if (result ne fix(DATA)) then test_status = FAIL
+
+# if (test_status eq PASS) then begin
+#     print, 'SUCCESS'
+# end else begin
+#     print, 'FAILURE'
+# endelse
+
+# ''',
+# '''
+
```

Comment:
Why comment out this test?

Reply:
See explanation given one post below (25-Oct-2023 11:58 a.m. PDT).